### DISCUSSION TITLE
Primi passi con l'autenticazione manuale (v2)

### DISCUSSION THREAD

**Discussion by sadjehwty** [fattureincloud/api#130] [Q&A] [answered]
Buongiorno,
sto facendo i primi esperimenti con v2 delle api
Ho provato ad usare l'autenticazione manuale, che sembra funzionare visto che sono riuscito ad ottenere il company_id tramite la chiamata /user/companies
Usando lo stesso token con l'id ottenuto al passo precedente mi aspettavo di poter ottenere un elenco dei clienti tramite la chiamata /c/{{company_id}}/entities/clients
Ma ottengo la risposta 
{
    "error": "unauthorized",
    "error_description": "The given user is not authorized to use this application."
}

Quando ho collegato l'applicazione ho messo privilegi completi nella sezione clienti quindi mi aspetterei di poter leggere e scrivere almeno quella sezione
Non ho aggiunto altri privilegi
Che cosa sbaglio?

Grazie

PS
le chiamate a /c/{{company_id}}/entities/clients le vedo in Mostra log con errore 401, quindi direi di non avere fatto confusione con il company_id

**Comment by valmoz** [comment, marked as answer]
Buongiorno @sadjehwty 
il messaggio di errore "The given user is not authorized to use this application." solitamente appare quando, utilizzando un'app privata, si cerca di utilizzare una company non appartenente all'utente che ha creato l'app stessa.
Solitamente questo problema si risolve inserendo tale utente nella whitelist dell'app privata, o rendendo l'app pubblica, ma in questo caso credo che il problema sia proprio il company id.
Le ricordo che deve sostituire il company id all'interno dell'URL, e che l'URL finale dovrebbe assumere una forma simile a:
`/c/123456/entities/clients`

Nel caso in cui il suo codice dovesse sembrarle ancora corretto, le consiglio di contattare il supporto clienti sulla pagina di Fatture in Cloud, inserendo anche il link a questa discussione. In questo modo potremo avere accesso anche a dati sensibili (come il company id) e potremo quindi effettuare degli accertamenti più approfonditi sul suo problema.

A disposizione per ulteriori domande
Mauro

**Reply by sadjehwty** [reply]
Perfetto, grazie, l'app era privata

**Reply by valmoz** [reply]
@sadjehwty è corretto che l'app sia privata, e puoi utilizzarla tranquillamente. L'app pubblica serve solo in casi ben definiti.
Puoi trovare nuove informazioni qui: [App Visibility](https://developers.fattureincloud.it/docs/basics/create-an-app#-app-visibility)
